File: src/main/java/model/item/Armor.java
package model.item;

import javafx.beans.property.IntegerProperty;
import javafx.beans.property.SimpleIntegerProperty;

public class Armor extends Equipment {

    private final IntegerProperty protection;
    private double weight;

    public Armor() {
        super();
        this.protection = new SimpleIntegerProperty(0);
        this.weight = 0;
    }

    public int getProtection() {
        return protection.get();
    }

    public IntegerProperty protectionProperty() {
        return protection;
    }

    public void setProtection(int protection) {
        this.protection.set(protection);
    }

    public double getWeight() {
        return weight;
    }

    public void setWeight(double weight) {
        this.weight = weight;
    }

    @Override
    public Armor copy() {
        Armor armor = (Armor) super.copy();
        armor.setProtection(this.getProtection());
        armor.setWeight(this.getWeight());

        return armor;
    }

    @Override
    public boolean equals(Object o) {

        if (!(o instanceof Armor) || !super.equals(o)) {
            return false;
        }

        Armor other = (Armor) o;

        return this.getProtection() == other.getProtection() && this.getWeight() == other.getWeight();
    }
}

File: app/src/main/java/de/pnp/manager/app/model/Armor.kt
package de.pnp.manager.app.model

import de.pnp.manager.model.ICurrency
import de.pnp.manager.model.IRarity
import de.pnp.manager.model.item.IArmor
import de.pnp.manager.model.item.IEquipment
import de.pnp.manager.model.upgrade.IUpgrade
import java.util.ArrayList

class Armor : IArmor, Equipment() {
    private var protection: Int = 0
    private var weight: Double = 0.0

    override fun copy(): IArmor {
        val armor = super.copy() as IArmor
        armor.protection = this.protection
        armor.weight = this.weight
        return armor
    }

    override fun getProtection(): Int {
        return this.protection
    }

    override fun setProtection(protection: Int) {
        this.protection = protection
    }

    override fun getProtectionWithWear(): Int {
        TODO("Not yet implemented")
    }

    override fun getWeight(): Double {
        return this.weight
    }

    override fun setWeight(weight: Double) {
       this.weight = weight
    }

}